### PR TITLE
feat(UploadField): support base64 (WH-1908)

### PR DESCRIPTION
**Pull request by tellmywife**
- add `encoding`  prop and remove `shouldBeBinary`
- update `onFileChange` function to handle `base64` encoding

### CODE REVIEW

File: lib/components/inputs/UploadField/UploadField.tsx [hunk 1]
```diff
@@ -17,19 +17,19 @@ interface UploadFieldProps {
   error?: string
   disabled?: boolean
   tooltipText?: string
-  shouldBeBinary?: boolean
+  encoding: 'text' | 'binary' | 'base64'
```
Comment (udia76) [inline lib/components/inputs/UploadField/UploadField.tsx:20]: Let's create constant ENCODING_TYPES or something, and use it in that file?

File: lib/components/inputs/UploadField/UploadField.tsx [hunk 2]
```diff
@@ -40,20 +40,34 @@ function UploadField(props: UploadFieldProps) {
     'upload-wrapper-disabled': disabled
   })
 
-  const onFileChange = (file) => {
-    if (shouldBeBinary) {
+  const onFileChange = (file: File) => {
+    if (encoding === 'binary') {
```
Comment (udia76) [inline lib/components/inputs/UploadField/UploadField.tsx:44]: We may want to add some empty file check, something like
```
if (file.size === 0) {
    onReadError?.(new Error('File is empty'))
    return
  }

File: lib/components/inputs/UploadField/UploadField.tsx [hunk 3]
```diff
@@ -40,20 +40,34 @@ function UploadField(props: UploadFieldProps) {
...
+
     try {
+      const isText = encoding === 'text'
       const reader = new FileReader()
-      reader.onload = (event) => {
+
+      reader.onload = (event: ProgressEvent<FileReader>) => {
         setFileName(file.name)
-        onChange(event.target.result)
+        const result = event.target?.result
```
Comment (udia76) [inline lib/components/inputs/UploadField/UploadField.tsx:55]: `const result = event.target?.result as string | null`

File: lib/components/inputs/UploadField/UploadField.tsx [hunk 4]
```diff
@@ -40,20 +40,34 @@ function UploadField(props: UploadFieldProps) {
...
+
```
Comment (udia76) [inline lib/components/inputs/UploadField/UploadField.tsx:56]: Maybe we need to verify result?
```
if (!result) {
        onReadError?.(new Error('Failed to read file content'))
        return
      }

File: lib/components/inputs/UploadField/UploadField.tsx [hunk 5]
```diff
@@ -40,20 +40,34 @@ function UploadField(props: UploadFieldProps) {
...
+          ? (result as string)
+          : (result as string).split(',')[1]
+
+        onChange(value)
       }
-      reader.readAsText(file)
+
+      reader.onerror = () => {
+        onReadError?.()
+      }
```
Comment (udia76) [inline lib/components/inputs/UploadField/UploadField.tsx:66]:
```
reader.onerror = (error) => {
      onReadError?.(error)
    }

File: lib/components/inputs/UploadField/UploadField.tsx [hunk 6]
```diff
@@ -40,20 +40,34 @@ function UploadField(props: UploadFieldProps) {
...
+
+      isText ? reader.readAsText(file) : reader.readAsDataURL(file)
     } catch (e) {
-      onReadError()
+      onReadError?.()
     }
```
Comment (udia76) [inline lib/components/inputs/UploadField/UploadField.tsx:71]:
```
catch (error) {
    onReadError?.(error as Error)
  }

Comment (tellmywife) [reply lib/components/inputs/UploadField/UploadField.tsx:71]: Thnx for review, will make updates.

Comment (tellmywife) [reply lib/components/inputs/UploadField/UploadField.tsx:79]: @udia76 I've made all updates, please take another look.

**Review by udia76 — APPROVED** [review]

**Comment by sofa-tata** [comment]
approving, just please don't forget to adapt wekapp accordingly, in has `shouldBeBinary` in `NFSKerberosAuthDialog`

**Review by sofa-tata — APPROVED** [review]

**Pull request merged**